reference/docs-conceptual/learn/ps101/06-flow-control.md

```diff
@@ -0,0 +1,336 @@
+>>     Get-ADComputer -Identity $Computer
+>> }
+```
+
+```Output
+DistinguishedName : CN=DC01,OU=Domain Controllers,DC=mikefrobbins,DC=com
+DNSHostName       : dc01.mikefrobbins.com
+Enabled           : True
+Name              : DC01
+ObjectClass       : computer
+ObjectGUID        : c38da20c-a484-469d-ba4c-bab3fb71ae8e
+SamAccountName    : DC01$
+SID               : S-1-5-21-2989741381-570885089-3319121794-1001
+UserPrincipalName :
+
+DistinguishedName : CN=WEB01,CN=Computers,DC=mikefrobbins,DC=com
+DNSHostName       : web01.mikefrobbins.com
+Enabled           : True
+Name              : WEB01
+ObjectClass       : computer
+ObjectGUID        : 33aa530e-1e31-40d8-8c78-76a18b673c33
+SamAccountName    : WEB01$
+SID               : S-1-5-21-2989741381-570885089-3319121794-1107
+UserPrincipalName :
+```
+
+Many times a loop such as `foreach` or `ForEach-Object` is necessary. Otherwise you'll receive an
+error message.
+
+```powershell
+Get-ADComputer -Identity 'DC01', 'WEB01'
+```
+
+```Output
+Get-ADComputer : Cannot convert 'System.Object[]' to the type
+'Microsoft.ActiveDirectory.Management.ADComputer' required by parameter 'Identity'.
+Specified method is not supported.
+At line:1 char:26
++ Get-ADComputer -Identity 'DC01', 'WEB01'
++                          ```````````````
+    + CategoryInfo          : InvalidArgument: (:) [Get-ADComputer], ParameterBindingExc
+   eption
+    + FullyQualifiedErrorId : CannotConvertArgument,Microsoft.ActiveDirectory.Management
+   .Commands.GetADComputer
+```
+
+Other times, you can get the same results while eliminating the loop altogether. Consult the cmdlet
+help to understand your options.
+
+```powershell
+'DC01', 'WEB01' | Get-ADComputer
+```
+
+```Output
+DistinguishedName : CN=DC01,OU=Domain Controllers,DC=mikefrobbins,DC=com
+DNSHostName       : dc01.mikefrobbins.com
+Enabled           : True
+Name              : DC01
+ObjectClass       : computer
+ObjectGUID        : c38da20c-a484-469d-ba4c-bab3fb71ae8e
+SamAccountName    : DC01$
+SID               : S-1-5-21-2989741381-570885089-3319121794-1001
+UserPrincipalName :
+
+DistinguishedName : CN=WEB01,CN=Computers,DC=mikefrobbins,DC=com
+DNSHostName       : web01.mikefrobbins.com
+Enabled           : True
+Name              : WEB01
+ObjectClass       : computer
+ObjectGUID        : 33aa530e-1e31-40d8-8c78-76a18b673c33
+SamAccountName    : WEB01$
+SID               : S-1-5-21-2989741381-570885089-3319121794-1107
+UserPrincipalName :
+```
+
+As you can see in the previous examples, the **Identity** parameter for `Get-ADComputer` only accepts a
+single value when provided via parameter input, but it allows for multiple items when the input is
+provided via pipeline input.
+
+### For
+
+A `for` loop iterates while a specified condition is true. The `for` loop is not something that I
+use often, but it does have its uses.
+
+```powershell
+for ($i = 1; $i -lt 5; $i++) {
+>>     Write-Output "Sleeping for $i seconds"
+>>     Start-Sleep -Seconds $i
+>> }
+```
```

Comment:
```suggestion
```powershell
for ($i = 1; $i -lt 5; $i++) {
  Write-Output "Sleeping for $i seconds"
  Start-Sleep -Seconds $i
}
```
```

Reply:
Fixed